TrailheadApex11/ExercisesPage/src/classes/Ex1SayingHello.cls

```diff
@@ -0,0 +1,18 @@
+/**
+ * Created by Kuba on 29.11.2017.
+ */
+
+public with sharing class Ex1SayingHello implements ExercisesLoader {
+    public String solveTheExercise(List<String> input) {
+        try {
+            String inputString = ExercisesController.isInputEmpty(input).get(0);
```

Comment:
(Tam gdzieś niżej napisałem, że będziesz miał NPE i że tam napiszę. Tutaj jest to 'tam' o którym mówiłem)

NPE - NullPointerException, często się używa NPE, w wielu językach jest taki sam, raczej zawsze wiadomo o co chodz.

Jeśli input jest serio empty to zrobisz .get(0) na pustej liście. Ja wiem, że try/catch, ale to nie zwalnia nas z konieczności nie bycia sloppy.

I w ogóle to nie powinno tu być try/catcha - z wyjątkami konwencja jest taka, że łapiesz je jak najwyżej (w tym przypadku w kontrolerze) a rzucasz jak najniżej - czyli tutaj. Nie zbierasz ich na każdym poziomie, bo wtedy będziesz je musiał wszędzie hendlować zamiast robić to raz. Do tego jak przejdziesz jakimś wywołaniem przez 7 klas i na dole rzucisz wyjątek to później masz śliczny stackTrace do prześledzenia co dokładnie poszło nie tak

Ta linijka jest najsłabsza w całym kodzie. Just fix that, pls ;P